### PR TITLE
Two more themes + Bump for release V1.22.

### LINKED ISSUE #57: Overview of additional colour themes (support?)
Hi again!
Much like @pserwylo, the arrival of a new baby in the house rekindled my interest in PianOli. (Congrats @pserwylo !)

I was happy to see all the activity, nice to see it's still going strong!
(the song player is very impressive)

One of the things that caught me out was the change of colour-scheme in #53 .
I agree that it's always better to follow "the" standard if there is one, but as the cynical saying goes "_the nice thing about standards is that there are so many to choose from_".

Personally, I'd never heard of "Boomwhackers", and had to search duckduckgo for them.  
To my layman's understanding, they seem to be rather USA-centric, rather than "international" as mentioned in the original PR.
Then again, there seems to be some software that supports it, and it doesn't hurt.

Again personally, I do miss the cheery brightness of the original rainbow. 

I'm opening this issue not as critique, but as a collection bucket for possible alternate standards, since the original MR mentioned the possibility of theme-switching.
 I'm aware of _at least_ one other variant, and an image search shows many different palettes.  
Perhaps other people know of more colour schemes?

# minimusiker 
In Germany, there is the "Minimusiker" convention, by a children-oriented band/musik-education team.  
This is a rainbow-like scheme, with a red C-note, .  

- They do sell lesson plans using this colour scheme:
  image  
  source: https://minimusiker.de/shop/Materialpaket-4-p253644213
  - note that the xylophone on the top-left corner has white C's, because they're not used in this rendition of "are you sleeping, brother John", the top-right shows the red C's.

I'm not sure how "big" that is in classrooms, or internationally, but they are at least somewhat influential in Germany among the general public.
- They partner with the (immensely popular, available in every children store) "[TonieBox](https://us.tonies.com/pages/toniebox)" music player, of which I (of course..) have one at home. 
- There are [mini-musiker-partnered xylophones](https://www.spiegelburg-shop.de/mein-allererstes-glockenspiel/63603) available in most bookstores, including [at Thalia](https://www.thalia.de/shop/home/artikeldetails/A1057907205) (the "has multiple shops in literally every German city" bookstore).
  - https://www.thalia.de/shop/home/artikeldetails/A1057907205
  - annoyingly, these use a _different_ colour scheme, with a _blue_ C, and adds in white for B. Because why keep your own standards...
    image
  - this of course speaks against it being an actual standard.

### PR DESCRIPTION
**Pull request by pserwylo**
Fixes #57.

Two themes added: Pastel + Rainbow. Refactored the way themes work to make it easier to define key colours.

This release will also include the dutch translations from #61.

### CODE REVIEW

File: app/src/main/java/com/nicobrailo/pianoli/Theme.java [hunk 1]
```diff
@@ -7,60 +7,98 @@
...
+        public final int colour;
+        public final int pressedColour;
+
+        public KeyColour(int colour, int pressedColour) {
+            this.colour = colour;
+            this.pressedColour = pressedColour;
+        }
+
+        public static KeyColour createLighterWhenPressed(int colour, float blendWhiteFactor) {
+            return new KeyColour(colour, ColorUtils.blendARGB(colour, Color.WHITE, blendWhiteFactor));
```
Comment (juleskers) [inline app/src/main/java/com/nicobrailo/pianoli/Theme.java:100]: Silly nitpick: should we always spell it "color", to to stay consistent with "ColorUtils"?

File: app/src/main/java/com/nicobrailo/pianoli/Theme.java [hunk 2]
```diff
@@ -7,60 +7,98 @@
...
 
     public int getColorForKey(int keyIndex, boolean isPressed) {
         final int col_idx = (keyIndex / 2) % colours.length;
-        return isPressed ? pressedColours[col_idx] : colours[col_idx];
+        final KeyColour key = colours[col_idx];
+        return isPressed ? key.pressedColour : key.colour;
+    }
+
+    private static class KeyColour {
+        public final int colour;
```
Comment (juleskers) [inline app/src/main/java/com/nicobrailo/pianoli/Theme.java:91]: Nitpick: "smurf naming";  
The `KeyColour`-class probably doesn't need to repeat "Colour" in each of its field names.
how about "normal" and "pressed"?

File: app/src/main/java/com/nicobrailo/pianoli/Theme.java [hunk 3]
```diff
@@ -7,60 +7,98 @@
...
+
+    private static final Theme RAINBOW = new Theme(
+            new KeyColour[] {
+                    KeyColour.createLighterWhenPressed(0xff001caf, 0.5f), // darkblue
+                    KeyColour.createLighterWhenPressed(0xff0099ff, 0.5f), // lightblue
+                    KeyColour.createLighterWhenPressed(0xff63c624, 0.5f), // darkgreen
+                    KeyColour.createLighterWhenPressed(0xffbde53d, 0.5f), // lightgreen
+                    KeyColour.createLighterWhenPressed(0xfffcc000, 0.5f), // yellow
+                    KeyColour.createLighterWhenPressed(0xffff810a, 0.5f), // lightorange
+                    KeyColour.createLighterWhenPressed(0xffff7416, 0.5f), // darkorange
```
Comment (juleskers) [inline app/src/main/java/com/nicobrailo/pianoli/Theme.java:66]: 💙 Nice helper method, really cleans up the repetitive blendARGB 👍

File: fastlane/metadata/android/en-US/images/phoneScreenshots/03_keys_pastel.png
Comment (juleskers) [inline fastlane/metadata/android/en-US/images/phoneScreenshots/03_keys_pastel.png:1]: This still looks like my "eyedropper" version. Did you end up not-using colorbrewer?

Definitely not a blocker, this is fine. Just curious what you ended up choosing.

**Review by juleskers — APPROVED** [review]
Looking good!

I have a minor nitpick over "colour"/"color", but other than that, looks mergeable to me! 👍

File: app/src/main/java/com/nicobrailo/pianoli/Theme.java (hunk 1 above)
Comment (pserwylo) [reply app/src/main/java/com/nicobrailo/pianoli/Theme.java]: Arg, the bane of my Australian existence! Happy to do so, I'm not precious about English spelling.

**Comment by pserwylo** [comment]
Thanks for the feedback as always @juleskers. Hope your first day back was fine. I'll tag this so F-Droid can build.

**Pull request merged**